### ISSUE TITLE
Icon not updating

### ISSUE THREAD

**Issue by anirudhb** [dfrankland/hyperterm-tab-icons#15]
Hyper 0.8.3
`hyper-tab-icons 1.1.2`

**How to reproduce**
1. Download a plugin like `hyperterm-title` or use the `git-HEAD` version of `Hyper.app`
2. Change the title to a custom title that is not the process name
3. Observe

When I start Python, or Node.js, or Ruby, the icon seems to stay as a shell.

Could this have something to do with me having a different title than the process name? If so, could there be a way to fix it?

**Comment by anirudhb** [comment]
How could you run a command _inside the current shell_ and get the output? This would help to find the icon.

**Comment by dfrankland** [comment]
I can't reproduce this on `Hyper.app@0.8.3` and `hyper-tab-icons@1.1.2`. There's no good way to know what the current running process is inside of another shell. This applies to things like the following:

*   `tmux`
*   `ssh`
*   Starting in one shell and switching to another like `sh`, `bash`, `fish`, or `zsh`.

**Comment by anirudhb** [comment]
Change the title with a plugin like `hyperterm-title` to a different title than the currently running process, then try.

**Comment by dfrankland** [comment]
I use `hyperterm-title` too. Make sure that `hyperterm-tab-icons` is after `hyperterm-title` like the following:

```js
module.exports = {
  config: { ... },
  plugins: [
    'hyperterm-title',  // Shows title suggested by shell, this runs first
    'hyper-tab-icons',  // Show icon for current process, this runs second and gets the result of `hyperterm-title`
  ],
};
```

**Comment by anirudhb** [comment]
Still doesn't work (with custom title)

**Comment by anirudhb** [comment]
Even vice-versa.

**Comment by dfrankland** [comment]
This merge breaks `hyperterm-title`: https://github.com/zeit/hyper/pull/892

Essentially `Hyper.app@0.8.4` will make `hyperterm-title` unneeded since the functionality will be built in 👍

**Comment by dfrankland** [comment]
So to fix this issue, if you are using `Hyper.app@HEAD` instead of the official `0.8.3` release please do not use `hyperterm-title`.

**Comment by anirudhb** [comment]
I'm not using `Hyper.app@HEAD`, I'm using 0.8.3

**Comment by dfrankland** [comment]
I am confused. Your "how to reproduce" in the initial issue says you are:
> **How to reproduce**
> 1.   Download a plugin like hyperterm-title or use the git-HEAD version of Hyper.app

If you are actually using `hyperterm-title` and `Hyper.app@0.8.3` (the official release version downloaded from [here](https://github.com/zeit/hyper/releases/tag/0.8.3)), I cannot reproduce the issue.

Is there any other information you could provide me?
*  What OS are you using?
*  What other plugins are you using?
*  Are you starting the app from the command line or from a shortcut?
*  Does `hyper-tab-icons@1.1.1` work still?
*  Does removing `~/.hyper_plugins/node_modules` and re-installing the dependencies again help?

**Comment by anirudhb** [comment]
It said 'or'

**Comment by anirudhb** [comment]
Starting from dock, macOS sierra, plugins list:
- `hyperterm-snazzy`
- `hyperterm-blink`
- `hyperterm-title`
- `hyperlinks`
- `hyperterm-cursor`
- `htyt`
- `hyper-tab-icons`

**Comment by anirudhb** [comment]
It was downloaded from hyper.is

**Comment by anirudhb** [comment]
Just to clarify: the icon **is showing up** but **it is not updating to the correct icon**

**Comment by anirudhb** [comment]
0.8.3.873

**Comment by dfrankland** [comment]
Using all of the same plugins, on macOS Sierra, using `Hyper.app@0.8.3.373`, launching from the toolbar. I still cannot reproduce this; all of the correct icons show up.

![](http://i.giphy.com/3oz8xXshBOLPRym9IA.gif)

**Comment by anirudhb** [comment]
Make sure the process name is not in the title.

**Comment by anirudhb** [comment]
Some code: `PROMPT_COMMAND='echo -ne "\033]0;blah\007"'`

**Comment by anirudhb** [comment]
With say, `PS1="$ "`

**Comment by anirudhb** [comment]
And no DEBUG trap

**Comment by dfrankland** [comment]
Ah, setting the title to an arbitrary string will break this plugin. This plugin is a presentational component only. See the ["How it Works" section in the README](https://github.com/dfrankland/hyper-tab-icons#how-it-works). All it does is look at the title provided by Hyper.app and shows an icon it thinks best matches it using fuzzy filtering.

**Comment by Arttse** [comment]
Same problem for me. Icon simply does not change. For purity, I'm use only this plugin.

![hyper-tab-icons](https://cloud.githubusercontent.com/assets/7123950/21511494/b7a6fb84-ccb2-11e6-914a-321410eb86af.gif)

---
Hyper `1.0.0`
hyper-tab-icons `1.1.2`
Linux Ubuntu `16.04`
Using `zsh` by default.

**Comment by dfrankland** [comment]
@Arttse, this is a slightly different issue. I've opened another issue for the problem you are seeing here: https://github.com/dfrankland/hyper-tab-icons/issues/17

Good catch btw!

**Comment by Arttse** [comment]
@dfrankland thanks for quick response! I thought that the problem is in the icon change, so I wrote here. It is not clear how it should look on Linux. Well, thanks again, I'll be waiting for improvements :+1:

**Comment by dfrankland** [comment]
This should be fixed with `hyper-tab-icons@1.1.3` now. See here: https://github.com/dfrankland/hyper-tab-icons#configtabiconsprocessnameregex
